lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -209,16 +210,22 @@ def self.is_firebasecmd_supported?(cmd)
       end
 
       def self.get_upload_token(app_id, binary_path)
-        binary_hash = Digest::SHA256.hexdigest(File.open(binary_path).read)
+        begin
+          binary_hash = Digest::SHA256.hexdigest(File.open(binary_path).read)
+        rescue Errno::ENOENT
+          UI.crash!(ErrorMessage::APK_NOT_FOUND)
+        end
+
         begin
           response = connection.get("#{PATH}#{app_id}") do |request|
             request.headers["Authorization"] = "Bearer " + auth_token
           end
         rescue Faraday::ResourceNotFound
-          UI.crash!("App Distribution could not find your app #{app_id}. Make sure to onboard your app by pressing the \"Get started\" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/_/appdistribution")
+          UI.crash!(ErrorMessage::INVALID_APP_ID)
```

Comment:
Consider including the `app_id` you tried in the error message, or even the full path.